### PR TITLE
refactor(login): don't hit login too often

### PR DESCRIPTION
**Pull request by hyrious**

### CODE REVIEW

**Review by Cheerego7 — DISMISSED** [review]

File: web/flat-web/src/pages/HomePage/index.tsx [hunk 1]
```diff
@@ -16,36 +16,42 @@ export const HomePage = observer(function HomePage() {
...
+                if (globalStore.lastLoginCheck) {
+                    if (Date.now() - globalStore.lastLoginCheck < 2 * 60 * 60 * 1000) {
+                        return true;
+                    }
+                }
                 try {
                     await loginCheck();
-                    nextPage = RouteNameType.HomePage;
+                    globalStore.lastLoginCheck = Date.now();
+                    isLoggedIn = true;
```
Comment (BlackHole1) [inline web/flat-web/src/pages/HomePage/index.tsx]: return true

File: web/flat-web/src/pages/HomePage/index.tsx [hunk 2]
```diff
@@ -16,36 +16,42 @@ export const HomePage = observer(function HomePage() {
...
                 } catch (e) {
+                    globalStore.lastLoginCheck = null;
                     console.error(e);
                     errorTips(e);
                 }
             }
 
-            return nextPage;
+            return isLoggedIn;
```
Comment (BlackHole1) [inline web/flat-web/src/pages/HomePage/index.tsx]: return fasle

File: web/flat-web/src/pages/HomePage/index.tsx [hunk 3]
```diff
@@ -16,36 +16,42 @@ export const HomePage = observer(function HomePage() {
...
     // eslint-disable-next-line react-hooks/exhaustive-deps
     useEffect(() => pageStore.configure(), []);
 
     useEffect(() => {
         let isUnMount = false;
 
-        async function checkLogin(): Promise<RouteNameType.HomePage | RouteNameType.LoginPage> {
-            let nextPage = RouteNameType.LoginPage;
+        async function checkLogin(): Promise<boolean> {
+            let isLoggedIn = false;
```
Comment (BlackHole1) [inline web/flat-web/src/pages/HomePage/index.tsx]: remove

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: web/flat-web/src/pages/HomePage/index.tsx (hunk 1 above)
Comment (BlackHole1) [reply web/flat-web/src/pages/HomePage/index.tsx]:
```typescript
const foo = () => {
  try {
    return 1;
  } catch(e) {}
}

foo() // result: 1
```

```typescript
const bar = () => {
  try {
    throw new Error("test");
  } catch(e) {
    return 2;
  }
}

bar() // result: 2
```

File: web/flat-web/src/pages/HomePage/index.tsx [hunk 4]
```diff
@@ -16,36 +16,40 @@ export const HomePage = observer(function HomePage() {
...
 
     useEffect(() => {
         let isUnMount = false;
 
-        async function checkLogin(): Promise<RouteNameType.HomePage | RouteNameType.LoginPage> {
-            let nextPage = RouteNameType.LoginPage;
-
+        async function checkLogin(): Promise<boolean> {
             const token = globalStore.userInfo?.token;
             if (token) {
```
Comment (BlackHole1) [inline web/flat-web/src/pages/HomePage/index.tsx]:
```typescript
if (!globalStore.userInfo?.token) {
  return false;
}
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

**Review by BlackHole1 — APPROVED** [review]

**Pull request merged**